#! /bin/sh -e
## 23_swarm-highmem-tlb.dpatch by Thiemo Seufer <seufer@csv.ica.uni-stuttgart.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Export missing module symbol

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

--- kernel-source-2.4.26/arch/mips/mm/tlb-sb1.c.old	2004-05-13 15:58:52.000000000 +0200
+++ kernel-source-2.4.26/arch/mips/mm/tlb-sb1.c	2004-05-13 15:59:49.000000000 +0200
@@ -18,6 +18,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
  */
 #include <linux/config.h>
+#include <linux/module.h>
 #include <linux/init.h>
 #include <asm/mmu_context.h>
 #include <asm/bootinfo.h>
@@ -238,6 +238,9 @@ void local_flush_tlb_one(unsigned long p
 	local_irq_restore(flags);
 }
 
+/* The highmem code wants this. */
+EXPORT_SYMBOL(local_flush_tlb_one);
+
 /* All entries common to a mm share an asid.  To effectively flush
    these entries, we just bump the asid. */
 void local_flush_tlb_mm(struct mm_struct *mm)
